### ISSUE TITLE
Users want to be able to load certificates from arbitrary secrets

### ISSUE THREAD

**Issue by kflynn** [datawire/ambassador#463]
Presently, Ambassador hardcodes certain Kubernetes secret names for TLS certificates. This is problematic both for TLS origination and for running multiple Ambassadors: a better way would be to allow Ambassador to specify which secret to load a given certificate from, e.g.:

```
apiVersion: ambassador/v0
kind:  Module
name:  tls
config:
  server:
    enabled: True
    secret: my-secret
```

or 

```
---
apiVersion: ambassador/v0
kind:  Mapping
name:  qotm_mapping
prefix: /qotm/
tls: 
  secret: qotm-upstream
service: qotm
```

or the like.

**Comment by containscafeine** [comment]
@kflynn this would mean abstracting out both the secrets, `ambassador-certs` and `ambassador-cacert`, right?

**Comment by containscafeine** [comment]
@kflynn how do you like -
```yaml
---
apiVersion: ambassador/v0
kind:  Module
name:  tls
config:
  server:
    enabled: True
    termination_secret: <secret name, optional>
    ca_secret: <secret name, optional>
```

**Comment by scher200** [comment]
@containscafeine clould you make me understand your total deployment.
In example case of two hello world projects of: https://github.com/datawire/hello-world-python.git
> from (https://forge.sh/)
Both hosted on a different domains, lets say: hello.datawire.io and hello.test.io
What kind of extra configs files would i need to add to roll this out to both of the projects?

**Comment by kflynn** [comment]
@containscafeine That looks reasonable, yeah.

@scher200 That's kind of adding scope beyond this issue -- more there in a bit. :)

**Comment by containscafeine** [comment]
Fixed in #526, released in 0.36.0 - github.com/datawire/ambassador/releases/latest
